### PR TITLE
MDEV-14798 Add, drop system versioning semantic and syntax

### PR DESCRIPTION
**Pull request by kevgs**

### CODE REVIEW

**Review by midenok — COMMENTED** [review]
`DROP PERIOD FOR SYSTEM_TIME` is subject to discussion under #389. We don't implement it right now, do we?

**Comment by kevgs** [comment]
Patch contained only syntax and semantic check for `DROP PERIOD FOR SYSTEM_TIME`, not the full feature as described in #389 . But I don't mind not implementing this right now. PR updated.

File: mysql-test/suite/versioning/r/alter.result [hunk 1]
```diff
@@ -125,7 +125,10 @@ with system versioning;
 select * from t for system_time all;
 a
 alter table t drop column sys_trx_start;
+ERROR HY000: Wrong parameters for `t`: missing 'DROP COLUMN sys_trx_end'
```
Comment (midenok) [inline mysql-test/suite/versioning/r/alter.result]: This is strange behavior.

File: mysql-test/suite/versioning/r/alter.result [hunk 2]
```diff
@@ -125,7 +125,10 @@ with system versioning;
...
 alter table t drop column sys_trx_end;
+ERROR HY000: Wrong parameters for `t`: missing 'DROP COLUMN sys_trx_start'
+alter table t drop column sys_trx_start, drop column sys_trx_end;
```
Comment (midenok) [inline mysql-test/suite/versioning/r/alter.result]: Dropping of system columns must be prohibited while SYSTEM VERSIONING is in effect. When SYSTEM VERSIONING is off, system columns become plain columns and can be dropped one-by-one. We don't have such option to turn off SYSTEM VERSIONING yet.

Comment (kevgs) [reply mysql-test/suite/versioning/r/alter.result]: Well, maybe you're right. But what to do with MDEV-14798? Suspend for now? Or commit other parts like invisibility requirement?

Comment (midenok) [reply mysql-test/suite/versioning/r/alter.result]: For this specific test case prohibit dropping of system columns. `DROP SYSTEM VERSIONING` does it as I suppose.

**Review by midenok — APPROVED** [review]
Rebase on latest trunk and merge.

**Pull request merged**